### ISSUE TITLE
Improve documentation and error validation for set_options(arithmetic_join)

### ISSUE THREAD

**Issue by crusaderky** [pydata/xarray#1624]
The documentation for set_options laconically says:

```
arithmetic_join: DataArray/Dataset alignment in binary operations. Default: 'inner'.
```

leaving the user wonder what the other options are.
Also, the set_options code does not make any kind of domain check on the possible values.
By scanning the code I gathered that the valid values (and their meanings) should be the same as align(join=...), but I'd like confirmation on that...

**Comment by shoyer** [comment]
> By scanning the code I gathered that the valid values (and their meanings) should be the same as align(join=...), but I'd like confirmation on that...

That's correct. We should update the documentation and add a check.

Using enums might have been a better idea here than string values, but that's somewhat rare in scipy/pandas code.

**Comment by rmsare** [comment]
This issue is pretty stale, but I've taken a crack at this. Not sure I like my approach. This doesn't seem like a high priority issue, but if we can triage my implementation with some feedback, happy to work it up into a PR.

[branch diff](https://github.com/pydata/xarray/compare/master...rmsare:check-arithmetic-join)

The idea was to add `_check_values` method to the options class that check the current values in `OPTIONS` against a dictionary which contains lists of accepted values. If there's no entry in that dictionary, any value is assumed valid. But values of incorrect type can be caught by a `_check_types` method, which checks for type consistency with the default values.

Both are called at the end of initialization to catch bad values before some function tries to use them in the new context.

Does xarray have any established value checking practices? Skimming through the code I couldn't find any.

**Comment by shoyer** [comment]
The validation was implemented in #2261, but the docstring could still use improvement.
